halibot/commandmodule.py

```diff
@@ -0,0 +1,59 @@
+#
+# CommandModule
+#    Subclass of HalModule that does the heavy lifting of commands for you
+#
+from . import HalModule
+
+# Decorator for converting the string stripped of the command to an argument list
+def AsArgs(func):
+	def wrapper(self, string, msg=None):
+		args = string.split(" ")
```

Comment:
This can give us empty arguments. Could filter out empties after the split, or trim and split over the regex `" +"`.

Reply:
Still thinking on this one. I agree that a regex would be better than `str.split()` in all cases in here. Only thing I am hung up on is, where to stick the compiled regex. There's no point in compiling it on every receive, so it should probably be a singleton somewhere?

Reply:
I don't think it really matters... you can just stick the regex in the top-level of the file.